internal/merger/merger.go

```diff
@@ -95,205 +89,159 @@ func MergeNode(local, remote *pbv2.NodeResponse) (*pbv2.NodeResponse, error) {
 			if err != nil {
 				return nil, err
 			}
-			remote.NextToken = updatedNextToken
-		}
-		return remote, nil
-	}
-
-	type linkedGraphStore struct {
-		propToNodes        map[string]*pbv2.Nodes
-		propToLinkedGraphs map[string]*pbv2.LinkedGraph
-		propSet            map[string]struct{}
-	}
-	dcidToLinkedGraph := map[string]*linkedGraphStore{}
-
-	collectNode := func(n *pbv2.NodeResponse) {
-		for dcid, linkedGrarph := range n.GetData() {
-			if _, ok := dcidToLinkedGraph[dcid]; !ok {
-				dcidToLinkedGraph[dcid] = &linkedGraphStore{}
-			}
-			if arcs := linkedGrarph.GetArcs(); len(arcs) > 0 {
-				if dcidToLinkedGraph[dcid].propToNodes == nil {
-					dcidToLinkedGraph[dcid].propToNodes = map[string]*pbv2.Nodes{}
-				}
-				for prop, nodes := range arcs {
-					dcidToLinkedGraph[dcid].propToNodes[prop] = nodes
+			aux.NextToken = updatedNextToken
+		}
+		return aux, nil
+	}
+	// Merge aux into main
+	mainData := main.GetData()
+	auxData := aux.GetData()
+	for dcid, linkedGraph := range auxData {
+		if mainData == nil {
+			mainData = map[string]*pbv2.LinkedGraph{}
+		}
+		if _, ok := mainData[dcid]; !ok {
+			mainData[dcid] = linkedGraph
+			continue
+		}
+		for prop, nodes := range linkedGraph.GetArcs() {
```

Comment:
nit: Any chance we create internal methods for some of this logic? It will help make this method shorter and easier to follow.

Reply:
Good point. Made a helper function for this.